pkg/model/gcemodel/autoscalinggroup.go

```diff
@@ -90,7 +90,7 @@ func (b *AutoscalingGroupModelBuilder) buildInstanceTemplate(c *fi.ModelBuilderC
 				BootDiskImage:  s(ig.Spec.Image),
 
 				// TODO: Support preemptible nodes?
-				Preemptible: fi.Bool(false),
+				Preemptible: fi.Bool(ig.Spec.GuestAccelerators != nil && len(ig.Spec.GuestAccelerators) > 0),
```

Comment:
Why is this related to preemption?

Reply:
Google Cloud forbids to create instances with GPUs in `MIGRATE` maintenance mode. Only `TERMINATE` is allowed.

Reply:
In that case, the nil check is not needed.

Reply:
fixed

Reply:
After reading a little, I think the correct way would be to set [Scheduling.OnHostMaintenance](https://pkg.go.dev/google.golang.org/api@v0.82.0/compute/v1#Scheduling.OnHostMaintenance).
`Preemptible ` seems to be the equivalent of Spot instances on AWS.

Reply:
Fixed. Moved to `InstanceTemplate.mapToGCE`